infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -38,12 +39,23 @@
                     .setNameFormat("LogUnit-Write-Processor-%d")
                     .build());
 
+    /**
+     * Barrier epoch persisted by the log unit.
+     */
+    private volatile long barrier;
```

Comment:
This comment is false and you need to describe what its used for. 
Also consider, EpochWaterMark.